packages/node-resolve/test/package-entry-points.js

```diff
@@ -352,7 +352,8 @@ test('can override a star pattern using null', async (t) => {
 
 test('can self-import a package when using exports field', async (t) => {
   const bundle = await rollup({
-    input: 'self-package-import',
+    // TODO previously this worked because we'd do this internally
+    input: './self-package-import',
```

Comment:
@guybedford this is needed now because the export map algo only runs on the import the user provided. So is this ok? If it is ok, does this make it a breaking change now? :/

Previously we 'fixed' `self-package-import` to `./self-package-import` and attempted that first

Reply:
If we have
```js
input: 'self-package-import2'
```
and rename `self-package-import.js` to `self-package-import2.js` it works fine.

Or if we just put `self-package-import.js` it also works, so going to switch it to that.

I think it fails now because the first run of the resolve algorithm is treating `self-package-import` as matching the package with that name over `./self-package-import.js`, which I think is correct?